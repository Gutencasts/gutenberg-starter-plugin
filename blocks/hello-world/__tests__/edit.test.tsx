import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HelloWorldBlock, { HelloWorldAttributes } from '../edit';
import '@testing-library/jest-dom';

jest.mock('@wordpress/block-editor', () => ({
  // Uncomment if you have a RichText field
  // RichText: jest.fn((props) => (
  //     <input
  //         value={props.value}
  //         placeholder={props.placeholder}
  //         onChange={(event: ChangeEvent<HTMLInputElement>): void => props.onChange(event?.target?.value)}
  //         data-testid={props['data-testid']}
  //     />
  // )),
  InspectorControls: jest.fn(({ children }) => <div>{children}</div>),
}));

let mockEditAttributes: HelloWorldAttributes;
const setAttributes = jest
  .fn()
  .mockImplementation((newAttributes: HelloWorldAttributes) => {
    mockEditAttributes = { ...mockEditAttributes, ...newAttributes };
  });

beforeEach(() => {
  setAttributes.mockClear();
  mockEditAttributes = {
    title: '',
  };
});

describe('Hello world block', () => {
  it('should update the title attribute', async () => {
    const {
      container, rerender, getByLabelText, getByText,
    } = render(<HelloWorldBlock
      isSelected
      clientId="hello-world-1"
      attributes={mockEditAttributes}
      setAttributes={setAttributes}
    />);

    const initialTitle = getByText('Hello world');

    expect(initialTitle).toBeInTheDocument();
    expect(initialTitle).toBeVisible();

    const titleField = getByLabelText('Title');

    expect(titleField).toBeVisible();

    await userEvent.type(titleField, 'My title');

    expect(setAttributes).toHaveBeenCalledWith({
      title: 'My title',
    });

    rerender(<HelloWorldBlock
      isSelected
      clientId="hello-world-1"
      attributes={mockEditAttributes}
      setAttributes={setAttributes}
    />);

    const title = getByText('Hello World from: My title');

    expect(title).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
