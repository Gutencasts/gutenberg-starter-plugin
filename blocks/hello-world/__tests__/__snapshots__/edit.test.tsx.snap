// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello world block should update the title attribute 1`] = `
<div>
  <div
    class="hello-world-block"
  >
    <h3
      class="hello-world__title"
    >
      Hello World from: My title
    </h3>
    <button
      class="btn btn-primary hello-world__ctx"
    >
      Click
    </button>
  </div>
  <div>
    <div
      class="components-panel__body is-opened"
    >
      <h2
        class="components-panel__body-title"
      >
        <button
          aria-expanded="true"
          class="components-button components-panel__body-toggle"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            <svg
              aria-hidden="true"
              class="components-panel__arrow"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.5 12.4L12 8l5.5 4.4-.9 1.2L12 10l-4.5 3.6-1-1.2z"
              />
            </svg>
          </span>
          Required fields
        </button>
      </h2>
      <div
        class="components-panel__row"
      >
        <div
          class="components-base-control"
        >
          <div
            class="components-base-control__field"
          >
            <label
              class="components-base-control__label"
              for="inspector-text-control-0"
            >
              Title
            </label>
            <input
              class="components-text-control__input"
              id="inspector-text-control-0"
              type="text"
              value="My title"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
